onnxruntime/core/providers/cpu/tensor/scatter.cc

```diff
@@ -339,8 +339,14 @@ Status ScatterData(
   //    output[i][indices[i][j][k]][k] = updates[i][j][k]
   // and so on
   std::vector<int64_t> dim_block_size(num_dims);
-
+#if defined(__GNUC__) && __GNUC__ >= 12
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"  // This warning occur when upgrading to gcc 12
+  dim_block_size.back() = 1;
+#pragma GCC diagnostic pop
```

Comment:
Change the assert to 
ORT_RETURN_IF_NOT(num_dims>0), you 
then this warning will be gone. I think. Because assert is only in debug mode, and you saw the warning in release mode. ORT_RETURN_IF_NOT works in both debug mode and release mode